File: src/com/app/test/util/DensityUtil.java
package com.app.test.util;


import android.content.Context;

import com.app.test.base.App;

/**
 *
 */
public class DensityUtil {

    private static float density;

    static {
        density = App.context.getResources().getDisplayMetrics().density;
    }

    public static int dp2px(int dp) {
        return (int) (dp * density + 0.5);
    }

    public static int px2dp(float px) {
        return (int) (px / density + 0.5f);
    }

    public static int px2sp(float px) {
        return (int) (px / density + 0.5f);
    }

    public static int sp2px(float sp) {
        return (int) (sp * density + 0.5f);
    }

    public static int getScreenHeight(Context context) {
        if (context != null) {
            return context.getResources().getDisplayMetrics().heightPixels;
        }
        return 0;
    }

    public static int getScreenWidth(Context context) {
        if (context != null) {
            return context.getResources().getDisplayMetrics().widthPixels;
        }
        return 0;
    }
}

File: src/com/app/test/util/DensityUtil.kt
package com.app.test.util

import android.content.Context
import com.app.test.base.App

/**
 *
 */
object DensityUtil {
    private var density = 0f

    @JvmStatic
    fun dp2px(dp: Int): Int {
        return (dp * density + 0.5).toInt()
    }

    @JvmStatic
    fun px2dp(px: Float): Int {
        return (px / density + 0.5f).toInt()
    }

    @JvmStatic
    fun px2sp(px: Float): Int {
        return (px / density + 0.5f).toInt()
    }

    @JvmStatic
    fun sp2px(sp: Float): Int {
        return (sp * density + 0.5f).toInt()
    }

    @JvmStatic
    fun getScreenHeight(context: Context?): Int {
        return context?.resources?.displayMetrics?.heightPixels ?: 0
    }

    @JvmStatic
    fun getScreenWidth(context: Context?): Int {
        return context?.resources?.displayMetrics?.widthPixels ?: 0
    }

    init {
        density = App.context?.resources?.displayMetrics?.density ?: 0f
    }
}